### PR TITLE
Add link to Threema Desktop 2.0 for iOS users

### PR DESCRIPTION
**Pull request by threema-gian**

### CODE REVIEW

File: src/partials/welcome.ts [hunk 1]
```diff
@@ -267,6 +267,24 @@ class WelcomeController {
...
+            const lastConnectedClientOS = localStorage.getItem("lastConnectedClientOS");
+            if (lastConnectedClientOS === "ios") {
+                return threema.OperatingSystem.Ios
+            } else if (lastConnectedClientOS === "android") {
+                return threema.OperatingSystem.Android
+            } else {
+                this.log.error("Unexpected value for lastConnectedClientOS was returned from localStorage");
+            }
+        } catch (error) {
+            this.log.error(`Could not write to localSotrage: ${error}`);
```
Comment (threema-danilo) [inline src/partials/welcome.ts:284]:
```suggestion
            this.log.error(`Could not read from localStorage: ${error}`);
```

File: src/partials/welcome.ts [hunk 2]
```diff
@@ -267,6 +267,24 @@ class WelcomeController {
...
+    public get lastConnectedClientOS(): threema.OperatingSystem | undefined {
+        try {
+            const lastConnectedClientOS = localStorage.getItem("lastConnectedClientOS");
+            if (lastConnectedClientOS === "ios") {
+                return threema.OperatingSystem.Ios
+            } else if (lastConnectedClientOS === "android") {
+                return threema.OperatingSystem.Android
+            } else {
+                this.log.error("Unexpected value for lastConnectedClientOS was returned from localStorage");
+            }
```
Comment (threema-danilo) [inline src/partials/welcome.ts:282]: If the local storage key is missing or empty, currently many errors are logged. In that case, I'd just return `undefined` without logging an error.

File: src/partials/welcome.html [hunk 3]
```diff
@@ -205,4 +205,10 @@ <h2 class="instructions" translate>welcome.ALREADY_CONNECTED</h2>
         </div>
 
     </div>
+
+    <a ng-if="!ctrl.showLoadingIndicator && ctrl.lastConnectedClientOS === 'ios'" class="desktop-beta-button" href="https://threema.ch/download-md">
+        <span class="badge">Beta</span>
+        <span translate>welcome.DESKTOP_BETA_HINT</span>
+        <md-icon class="material-icons md-24">arrow_forward</md-icon>
+    </a>
```
Comment (threema-danilo) [inline src/partials/welcome.html:213]: If you set a password, then the badge is offset from the bottom:

screenshot-20240611-181744

That's probably not intended, right?

File: src/partials/welcome.html [hunk 4]
```diff
@@ -205,4 +205,10 @@ <h2 class="instructions" translate>welcome.ALREADY_CONNECTED</h2>
         </div>
 
     </div>
+
+    <a ng-if="!ctrl.showLoadingIndicator && ctrl.lastConnectedClientOS === 'ios'" class="desktop-beta-button" href="https://threema.ch/download-md">
```
Comment (threema-danilo) [inline src/partials/welcome.html:209]: When scanning the code, the badge remains visible for about half a second. Could probably be improved by better positioning (as discussed).

**Review by threema-danilo — CHANGES_REQUESTED** [review]

File: src/partials/welcome.ts (hunk 2 above)
Comment (threema-gian) [reply src/partials/welcome.ts:282]: Amended!

File: src/partials/welcome.html (hunk 3 above)
Comment (threema-gian) [reply src/partials/welcome.html:213]: No, that's definitely incorrect. Thanks! Amended.

File: src/partials/welcome.html (hunk 4 above)
Comment (threema-gian) [reply src/partials/welcome.html:209]: Amended!

File: public/i18n/de.json [hunk 5]
```diff
@@ -5,8 +5,12 @@
     "welcome": {
         "ABORT": "Abbrechen",
         "CONTINUE_ANYWAY": "Trotzdem fortfahren",
+        "DESKTOP_BETA_HINT": "Nutzen Sie die neue Desktop-App",
+        "DESKTOP_BETA_HINT_INTRO_TEXT1": "Nutzen Sie bereits die neue Threema Desktop-App?",
+        "DESKTOP_BETA_HINT_INTRO_TEXT2": "Laden Sie die App gleich herunter.",
         "PLEASE_SCAN": "Scannen Sie den QR-Code mit Ihrer Threema-App",
-        "PLEASE_SCAN_INSTRUCTIONS": "Android: Menu > Desktop/Web | iOS: Einstellungen > Desktop/Web",
+        "PLEASE_SCAN_INSTRUCTIONS_ANDROID": "Android: Menu > Desktop/Web",
+        "PLEASE_SCAN_INSTRUCTIONS_IOS": "iOS: iOS: Einstellungen > Desktop/Web",
```
[the remark this answers is not in the document]
Comment (threema-danilo) [reply public/i18n/de.json]: Thanks for the hint 🙂

**Review by threema-danilo — APPROVED** [review]
👌🏻

**Pull request merged**